x4.md

```diff
@@ -54,325 +49,93 @@ The protocol should be a reflection of our principles.
 
-1. Calculate a symmetric key using [`shh_generateSymKeyFromPassword`](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_generatesymkeyfrompassword) passing a public chat name as a string and save the result to `symKeyID`
-2. Call [`shh_post`](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_post) with the following settings:
-   1. `symKeyID` MUST be set to `symKeyID`
-   2. `sig` MUST be set to `sigKeyID`
-   3. `ttl` MUST be at least `10` (it is in seconds)
-   4. `topic` MUST be set accordingly to [Topic](#topic) section and hex-encoded,
-   5. `payload` MUST be a hex-encoded string,
-   6. `powTime` MAY be arbitrary but should be enough to perform proof-of-work
-   7. `powTarget` MUST be equal or lower than `0.002`.
+# Security layer
 
-Learn more following [Whisper V6 RPC API](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API).
+Security layer provides various cryptographical properties to the Status Protocol:
+* encryption
+* assurance of the origin
+* forward secrecy
 
-## Receiving
+Encryption makes sure that only author and the recipient can read the content of the message.
 
-Receiving public messages depends on Whisper filters idea. Upon receiving, messages are first matched by a topic and then trying to be decrypted using a symmetric key.
+Assurance of the origin using a digital signature ensures that the message was actually sent by the owner of a given public key and the message has not been changed induring the transport over the wire.
 
-1. Calculate a symmetric key using [`shh_generateSymKeyFromPassword`](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_generatesymkeyfrompassword) passing public chat name as a string and save the result to `symKeyID`
-2. Call [`shh_subscribe`](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_subscribe) with criteria:
-   1. `minPow` MUST be at least `0.002`
-   2. `topics` MUST be list of hex-encoded topics you expect messages to receive from (follow [Topic](#topic) section)
-   3. `allowP2P` MUST be set to `true` if offline messages are supported, otherwise can be `false`.
+Note: exchange and verification of the public keys is not a trivial problem. Read more about this problem in [Initial Trust Establishment Specification](x5.md).
 
-Alternative method is to use [`shh_newMessageFilter`](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_newmessagefilter) which takes the same criteria object and then periodically calling [`shh_getFilterMessages`](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_getfiltermessages) method.
+Note: The Status Procol can work with PFS enabled and disabled. In the case of disabled PFS, the encryption is moved to the Privacy Layer and handled by Whisper. Whisper does **not** provide forward secrecy. This is a flaw of the original design which mixes security and private layer responsibilities.
```

Comment:
Typo

Is it moved or is it encrypted double (or single)?

Reply:
It's encrypted twice when PFS is used. When PFS is disabled there is no encryption in the  Conversational Security layer. 

This fragment is about a case when PFS is disabled. My point here was that the responsibility to encrypt the payload and provide confidentiality, integrity etc. is entirely on the Privacy Layer then.